src/resolve/resolved-dns-dnssec.c

```diff
@@ -1185,6 +1186,10 @@ int dnssec_verify_rrset_search(
                 DnsResourceRecord *dnskey;
                 DnsAnswerFlags flags;
 
+                /* Have we seen an unreasonable number of invalid signaures? */
+                if (nvalidations > DNSSEC_INVALID_MAX)
```

Comment:
i think conceptually it would make a ton of sense to have a check like this also in the inner loop, i.e. immediately before we call dnssec_verify_rrset() because that's where we actually waste the CPU cycles.

and I think it would make sense to return DNSSEC_TOO_MANY_VALIDATIONS in *result here already if this is hit, because it's the most interesting error i'd say, it should trump the others I think. 

hence maybe just move this check down, right before dnssec_verify_rrset() and make it:

```c
if (nvalidations > DNSSEC_INVALID_MAX) {
        /* Too many invalid validations trumps everything. Return immediately */
        *result = DNSSEC_TOO_MANY_VALIDATIONS;
        return 0;
}
```

or so?

Reply:
I had it basically like this at some point then changed it because I thought it looked cleaner that way. But actually yeah I think this is probably better.